### PR TITLE
fix(MeshHelpers): add CurveMerger

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/mesh/helpers/detail/curve_merger.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (panquez) [inline include/geode/mesh/helpers/detail/curve_merger.h:1]: 2023

**Review by panquez — APPROVED** [review]

File: include/geode/mesh/helpers/detail/curve_merger.h (hunk 1 above)
Comment (BotellaA) [reply include/geode/mesh/helpers/detail/curve_merger.h:1]: nice catch!

File: include/geode/mesh/helpers/private/vertex_merger.h [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            {
+                auto info = create_colocated_index_mapping();
+                vertices_ = std::move( info.colocated_mapping );
+                for( const auto& point : info.unique_points )
+                {
+                    builder_->create_point( point );
+                }
+            }
+
+            ColocatedInfo create_colocated_index_mapping()
```
Comment (MelchiorSchuh) [inline include/geode/mesh/helpers/private/vertex_merger.h:127]: tu peux mettre cette fonction en private

File: src/geode/mesh/helpers/detail/curve_merger.cpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                curve_id_.reserve( nb_edges );
+            }
+
+            std::unique_ptr< EdgedCurve< dimension > > merge()
+            {
+                create_curve_step();
+                return this->steal_mesh();
+            }
+
+            void create_curve_step()
```
Comment (MelchiorSchuh) [inline src/geode/mesh/helpers/detail/curve_merger.cpp:74]: I think you can pass this as private

File: src/geode/mesh/helpers/detail/curve_merger.cpp [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+                            new_id_[s][e] = old2new[old];
+                        }
+                    }
+                }
+            }
+
+            void create_edges()
+            {
+                absl::flat_hash_map< TypedVertexCycle, index_t > edges;
+                for( const auto s : Indices{ this->meshes() } )
```
Comment (MelchiorSchuh) [inline src/geode/mesh/helpers/detail/curve_merger.cpp:133]: I don't like one letter variables if they're used more than one time in the following.

File: src/geode/mesh/helpers/detail/surface_merger.cpp [hunk 5]
```diff
@@ -98,54 +69,32 @@ namespace geode
             std::unique_ptr< SurfaceMesh< dimension > > merge()
             {
                 create_surface_step();
-                return std::move( surface_ );
+                return this->steal_mesh();
             }
 
             void create_surface_step()
```
Comment (MelchiorSchuh) [inline src/geode/mesh/helpers/detail/surface_merger.cpp:75]: You can pass this as private

**Review by MelchiorSchuh — APPROVED** [review]

**Pull request merged**